libzfs.pyx

```diff
@@ -651,14 +652,13 @@ cdef class ZFS(object):
             if top_level_prop not in all_props:
                 raise ValueError(f'{top_level_prop} should be present in props.')
 
-        for p in self.pools:
+        for ds_name in datasets:
             c_name = handle = NULL
-            name = p.name
-            c_name = name
+            c_name = ds_name
 
             dataset = [
                 {
-                    'pool': name,
+                    'pool': ds_name.split('/')[0],
```

Comment:
ds_name.split('/', 1)[0]

We can save 0.0001 joules by not splitting subdatasets :P

Reply:
Agreed ; )